### ISSUE TITLE
NewRelic Module cannot be removed

### ISSUE THREAD

**Issue by IbrahimS2** [magento/magento2#35479]
<!---
Please review our guidelines before adding a new issue: https://github.com/magento/magento2/wiki/Issue-reporting-guidelines
Fields marked with (*) are required. Please don't remove the template.
-->

### Preconditions (*)
<!---
Provide the exact Magento version (example: 2.4.0) and any important information on the environment where bug is reproducible.
-->
1. M2.4.4
2. PHP8.1

### Steps to reproduce (*)
<!---
Important: Provide a set of clear steps to reproduce this bug. We can not provide support without clear instructions on how to reproduce.
-->
1. Remove NewRelic Entirely from composer
2. Recompile the code

### Expected result (*)
<!--- Tell us what do you expect to happen. -->
1. All goes smooth, no issues

### Actual result (*)
<!--- Tell us what happened instead. Include error messages and issues. -->
1. 
```
Area configuration aggregation... 5/9 [===============>------------]  55% 39 secs 370.0 MiB
In ClassReader.php line 57:
                                                                                                                                      
  Impossible to process constructor argument Parameter #0 [ <required> Magento\NewRelicReporting\Model\Config $config ] of Magento\   
  GraphQl\Model\Query\Logger\NewRelic class                                                                                           
                                                                                                                                      

In GetParameterClassTrait.php line 34:

  Class "Magento\NewRelicReporting\Model\Config" does not exist
```

---
Please provide [Severity](https://devdocs.magento.com/guides/v2.4/contributor-guide/contributing.html#backlog) assessment for the Issue as Reporter. This information will help during Confirmation and Issue triage processes.

- [ ] Severity: **S0** _- Affects critical data or functionality and leaves users without workaround._
- [X] Severity: **S1** _- Affects critical data or functionality and forces users to employ a workaround._
- [ ] Severity: **S2** _- Affects non-critical data or functionality and forces users to employ a workaround._
- [ ] Severity: **S3** _- Affects non-critical data or functionality and does not force users to employ a workaround._
- [ ] Severity: **S4** _- Affects aesthetics, professional look and feel, “quality” or “usability”._

**Comment by 1ci** [comment]
Please fix this. I can't switch the deploy mode to production otherwise and the store is super slow.

**Comment by IbrahimS2** [comment]
That is the thing; we do not want to have NewRelic entirely.

**Comment by engcom-Dash** [comment]
@magento give me 2.4-develop instance

**Comment by engcom-Dash** [comment]
@magento give me 2.4-develop instance

**Comment by engcom-Dash** [comment]
@magento give me 2.4-develop instance

**Comment by engcom-Dash** [comment]
Hi @IbrahimS2 ,
Can you please guide how to remove newrelic module using composer. Because we are finds the same in composer.json file in magento 2.4.4

Thanks

**Comment by hostep** [comment]
@engcom-Dash: add the following lines to your `composer.json` file:

```
    "replace": {
        "magento/module-new-relic-reporting": "*"
    }
```

Then execute `composer update magento/module-new-relic-reporting` (which should remove that package) and then execute `bin/magento setup:upgrade` (so it gets removed from the `app/etc/config.php` file).



I agree with the issue report, a dependency on the NewRelicReporting module should have never been made to the GraphQl module, that's bad design!
This was introduced in Magento 2.4.4 in scope of [PWA-1311: New Relic is not being given useful transaction names for graphql requests](https://github.com/magento/magento2/commit/1483c7a45ca1c2e82302353078b672610fc3c524)
It would have been better if a new module got created to connect graphql with new-relic-reporting, that would have been a much cleaner solution.

**Comment by engcom-Dash** [comment]
Hi @hostep ,
The issue Confirmed and we are also Getting Same results.
Here i Attached the Screenshot For your reference
![Screenshot from 2022-06-17 12-56-28](https://user-images.githubusercontent.com/60198592/174248117-c41f335c-b448-4c94-ad61-668dd99db221.png)

Thanks.

**Comment by sprankhub** [comment]
I suggest that the Magento core team uses https://github.com/yireo/Yireo_ExtensionChecker/ to make sure all dependencies are found :see_no_evil:

**Comment by WebKenth** [comment]
We use GraphQL but not NewRelic, why the deep coupling between the two modules i have no clue, but here's my cent

main issue might be:
vendor/magento/module-graph-ql/etc/module.xml:12

Missing module sequence `<module name="Magento_NewRelicReporting"/>`

If you want to patch it while waiting for the core team to figure out how to resolve it:

You can use this patch:
It removes the di.xml reference to the logger module and empties the `vendor/magento/module-graph-ql/Model/Query/Logger/NewRelic.php` file 

```patch
--- etc/di.xml
+++ etc/di.xml
@@ -104,11 +104,4 @@
             <argument name="queryComplexity" xsi:type="number">300</argument>
         </arguments>
     </type>
-    <type name="Magento\GraphQl\Model\Query\Logger\LoggerPool">
-        <arguments>
-            <argument name="loggers" xsi:type="array">
-                <item name="newRelic" xsi:type="object">Magento\GraphQl\Model\Query\Logger\NewRelic</item>
-            </argument>
-        </arguments>
-    </type>
 </config>
--- Model/Query/Logger/NewRelic.php
+++ Model/Query/Logger/NewRelic.php
@@ -1,56 +0,0 @@
-<?php
-/**
- * Copyright © Magento, Inc. All rights reserved.
- * See COPYING.txt for license details.
- */
-
-namespace Magento\GraphQl\Model\Query\Logger;
-
-use Magento\NewRelicReporting\Model\Config;
-use Magento\NewRelicReporting\Model\NewRelicWrapper;
-
-/**
- * Logs GraphQl query data for New Relic
- */
-class NewRelic implements LoggerInterface
-{
-    /**
-     * @var Config
-     */
-    private $config;
-
-    /**
-     * @var NewRelicWrapper
-     */
-    private $newRelicWrapper;
-
-    /**
-     * @param Config $config
-     * @param NewRelicWrapper $newRelicWrapper
-     */
-    public function __construct(
-        Config $config,
-        NewRelicWrapper $newRelicWrapper
-    ) {
-        $this->config = $config;
-        $this->newRelicWrapper = $newRelicWrapper;
-    }
-
-    /**
-     * @inheritdoc
-     */
-    public function execute(array $queryDetails)
-    {
-        if (!$this->config->isNewRelicEnabled()) {
-            return;
-        }
-
-        foreach ($queryDetails as $key => $value) {
-            $this->newRelicWrapper->addCustomParameter($key, $value);
-        }
-
-        $transactionName = $queryDetails[LoggerInterface::OPERATION_NAMES] ?: '';
-
-        $this->newRelicWrapper->setTransactionName('GraphQL-' . $transactionName);
-    }
-}
```

I've also added the replace on the new relic module to remove it completely from composer
```
"replace": {
...
        "magento/module-new-relic-reporting": "*"
...
}
```

**Comment by hostep** [comment]
Is probably fixed internally by https://github.com/magento/magento2/commit/a495fed6e678da185a74de697c946e5637a781a1

**Comment by hostep** [comment]
Confirmed with Magento 2.4.7-beta3, we can remove the `magento/module-new-relic-reporting` module again via composer and have a working `setup:di:compile`

FYI, you should probably also remove these 2 new modules that come with Magento 2.4.7 if you're not interested in New Relic:
- `magento/module-application-performance-monitor-new-relic`
- `magento/module-graph-ql-new-relic`

**Comment by tuyennn** [comment]
Still persist on Magento 2.4.7

**Comment by hostep** [comment]
@tuyennn : No it doesn't, it seems fixed in 2.4.7, see [my previous comment](https://github.com/magento/magento2/issues/35479#issuecomment-2034424156)...

This issue can be closed in my opinion.

**Comment by tuyennn** [comment]
@hostep agreed, that seems I applied the yireo replacement so find out this issue.
